job/server/src/main/java/alluxio/job/util/JobUtils.java

```diff
@@ -128,13 +129,32 @@ public static void loadBlock(URIStatus status, FileSystemContext context, long b
       throw new AlluxioException(
           ExceptionMessage.PINNED_TO_MULTIPLE_MEDIUMTYPES.getMessage(status.getPath()));
     }
+
+    if (pinnedLocation.isEmpty()) {
+      OpenFilePOptions openOptions =
+          OpenFilePOptions.newBuilder().setReadType(ReadPType.CACHE_PROMOTE).build();
+      InStreamOptions inOptions = new InStreamOptions(status, openOptions, conf);
+      BlockInfo info = status.getBlockInfo(blockId);
+      if (info == null) {
+        // we shall not reach here
+        throw new AlluxioException("Invalid block Id " + blockId + ", status " + status);
+      }
+      try (InputStream inputStream = BlockInStream.create(context, info, localNetAddress,
```

Comment:
Use `BlockInStream.create` will further save one RPC compared to `blockStore.getOutStream` which fetches `blockInfo` from master